File: app/src/main/java/com/sorcerer/sorcery/iconpack/util/ViewUtil.java
package com.sorcerer.sorcery.iconpack.util;

import android.view.View;

/**
 * @description:
 * @author: Sorcerer
 * @date: 2016/4/27
 */
public class ViewUtil {
    public static boolean isPointInsideView(float x, float y, View view) {
        int location[] = new int[2];
        view.getLocationOnScreen(location);
        int viewX = location[0];
        int viewY = location[1];

        return (x > viewX && x < (viewX + view.getWidth()))
                && (y > viewY && y < (viewY + view.getHeight()));
    }

    public static boolean isClick(float startX, float endX, float startY, float endY) {
        float differenceX = Math.abs(startX - endX);
        float differenceY = Math.abs(startY - endY);
        return !(differenceX > 5 || differenceY > 5);
    }

}

File: app/src/main/java/com/sorcerer/sorcery/iconpack/utils/ViewUtil.kt
package com.sorcerer.sorcery.iconpack.utils

import android.view.View

/**
 * @description:
 * *
 * @author: Sorcerer
 * *
 * @date: 2016/4/27
 */
object ViewUtil {
    fun isPointInsideView(x: Float, y: Float, view: View): Boolean {
        val location = IntArray(2)
        view.getLocationOnScreen(location)
        val viewX = location[0]
        val viewY = location[1]

        return x > viewX && x < viewX + view.width && y > viewY && y < viewY + view.height
    }

    fun isClick(startX: Float, endX: Float, startY: Float, endY: Float): Boolean {
        val differenceX = Math.abs(startX - endX)
        val differenceY = Math.abs(startY - endY)
        return !(differenceX > 5 || differenceY > 5)
    }

}
